src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -705,10 +714,29 @@ ucs_status_t uct_rc_mlx5_devx_iface_subscribe_event(
         struct mlx5dv_devx_obj *obj, uint16_t event, uint64_t cookie,
         char *msg_arg);
 
+#if HAVE_DEVX
 ucs_status_t
 uct_rc_mlx5_devx_iface_init_events(uct_rc_mlx5_iface_common_t *iface);
 
 void uct_rc_mlx5_devx_iface_free_events(uct_rc_mlx5_iface_common_t *iface);
+#else
+static UCS_F_MAYBE_UNUSED ucs_status_t
+uct_rc_mlx5_devx_iface_init_events(uct_rc_mlx5_iface_common_t *iface)
+{
+    if (uct_ib_mlx5_iface_md(&iface->super.super)->flags &
```

Comment:
maybe remove this check, also remove it from uct_rc_mlx5_devx_iface_init_events in C file, and check the UCT_IB_MLX5_MD_FLAG_DEVX condition before calling uct_rc_mlx5_devx_iface_init_events from uct_rc_mlx5_iface_common_t constructor?

Reply:
Done